api/xds_protocol.rst

```diff
@@ -356,21 +365,30 @@ After a NACK, an API update may succeed at a new version **Y**:
 .. figure:: diagrams/later-ack.svg
    :alt: ACK after NACK
 
+The preferred mechanism for a server to detect a NACK is to look for the presence of the
+:ref:`error_detail <envoy_api_field_DiscoveryRequest.error_detail>` field in the request sent by
+the client. Some older servers may instead detect a NACK by looking at both the version and the
+nonce in the request: if the version in the request is lower than the one sent by the server with
+that nonce, then the client has rejected the most recent version. However, this approach does not
+work for APIs other than LDS and CDS for clients that may dynamically change the set of resources
+that they are subscribing to, unless the server has somehow arranged to increment the resource
+type instance version every time any one client subscribes to a new resource.
+
 ACK and NACK semantics summary
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 - The xDS client should ACK or NACK every :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
-  received from the management server.
-- Like all other requests, the nonce from the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
-  is sent as :ref:`response_nonce <envoy_api_field_DiscoveryRequest.response_nonce>`.
-  As described in :ref:`resource update <xds_protocol_resource_update>` the nonce is
-  used in certain race conditions to disambiguate between ACK and NACK.
+  received from the management server. The :ref:`response_nonce
+  <envoy_api_field_DiscoveryRequest.response_nonce>` field tells the server which of its responses
+  the ACK or NACK is associated with.
 - ACK signifies successful configuration update and contains the
   :ref:`version_info <envoy_api_field_DiscoveryResponse.version_info>` from the
   :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`.
-- NACK signifies unsuccessful configuration update and contains the previous (existing)
-  :ref:`version_info <envoy_api_field_DiscoveryResponse.version_info>`.
-- Only the NACK should populate the :ref:`error_detail <envoy_api_field_DiscoveryRequest.error_detail>`.
+- NACK signifies unsuccessful configuration and is indicated by the presence of the
+  :ref:`error_detail <envoy_api_field_DiscoveryRequest.error_detail>` field.  The :ref:`version_info
+  <envoy_api_field_DiscoveryResponse.version_info>` indicates the most recent version that the
+  client is using, although that may not be an older version in the case where the client has
+  subscribed to a new resource from an existing version and that new resource is invalid.
```

Comment:
This last sentence is a bit unclear. I think I get at what you mean based on our earlier conversation but not sure other readers will.

Reply:
To clarify this, I've added an example above and referred back to it here.